File: cifrado/escritorio/codigo/Ruby/procesar/Cifrador.rb
class Cifrador
    def initialize(alfabeto)
            @alfabeto = alfabeto
    end

    def cifrarCaracter(recorrido,caracter)
        posicion = @alfabeto.obtenerPosicion(caracter)
        if posicion > 0
            return @alfabeto.obtenerCaracter(posicion+recorrido)
        else
            return caracter
        end
    end

    def cifrarTexto(recorrido,texto)
        textoCifrado = ""
        texto.chars.each{
            |caracter| textoCifrado = textoCifrado + cifrarCaracter(recorrido,caracter)
        }
        return textoCifrado
    end
end

File: cifrado/escritorio/codigo/Kotlin/src/procesar/Cifrador.kt
package procesar
import alfabeto.Alfabeto
import kotlin.properties.Delegates

class Cifrador(alfabeto: Alfabeto){
    var alfabeto : Alfabeto by Delegates.notNull()

    init {
        this.alfabeto = alfabeto
    }

    fun cifrarCaracter(recorrido:Int,caracter:Char): Char {
        val posicion = alfabeto.obtenerPosicion(caracter)
        if (posicion > 0)
            return alfabeto.obtenerCaracter(posicion+recorrido)
        else
            return caracter
    }

    fun cifrarTexto(recorrido:Int,texto:String): String {
        var textoCifrado = ""
        texto.toCharArray().forEach {
            textoCifrado = textoCifrado + cifrarCaracter(recorrido,it)

        }

        return textoCifrado
    }
}